modules/protocol-tests/test/src/smithy4s/api/validation/AddMemberTraitValidatorSpec.scala

```diff
@@ -0,0 +1,224 @@
+    val adtTrait = new AdtMemberTrait(unionShapeId)
+    val structMember = MemberShape
+      .builder()
+      .id("test#struct$testing")
+      .target("smithy.api#String")
+      .build()
+    val struct =
+      StructureShape
+        .builder()
+        .id("test#struct")
+        .addTrait(adtTrait)
+        .addMember(structMember)
+        .build()
+
+    val unionMember = MemberShape
+      .builder()
+      .id(unionShapeId.withMember("unionMember"))
+      .target(struct.getId)
+      .build()
+    val union =
+      UnionShape.builder().id(unionShapeId).addMember(unionMember).build()
+
+    val union2ShapeId = ShapeId.fromParts("test", "MyUnionTwo")
+    val unionMember2 = unionMember.toBuilder
+      .id(union2ShapeId.withMember("unionMemberTwo"))
+      .build()
+    val union2 =
+      UnionShape.builder().id(union2ShapeId).addMember(unionMember2).build()
+
+    val model =
+      Model.builder().addShapes(struct, union, union2).build()
+
+    val result = validator.validate(model).asScala.toList
+
+    val expected = List(
+      ValidationEvent
+        .builder()
+        .id("AdtMemberTrait")
+        .shape(struct)
+        .severity(Severity.ERROR)
+        .message(
+          "test#struct is improperly referenced from [test#MyUnionTwo]"
+        )
+        .build()
+    )
+    expect(result == expected)
+  }
+
+  test("return error when structure is targeted by a union and a structure") {
+    val unionShapeId = ShapeId.fromParts("test", "MyUnion")
+    val adtTrait = new AdtMemberTrait(unionShapeId)
+    val structMember = MemberShape
+      .builder()
+      .id("test#struct$testing")
+      .target("smithy.api#String")
+      .build()
+    val struct =
+      StructureShape
+        .builder()
+        .id("test#struct")
+        .addTrait(adtTrait)
+        .addMember(structMember)
+        .build()
+
+    val unionMember = MemberShape
+      .builder()
+      .id(unionShapeId.withMember("unionMember"))
+      .target(struct.getId)
+      .build()
+    val union =
+      UnionShape.builder().id(unionShapeId).addMember(unionMember).build()
+
+    val struct2ShapeId = ShapeId.fromParts("test", "MyStruct2")
+    val structMember2 = unionMember.toBuilder
+      .id(struct2ShapeId.withMember("structMember2"))
+      .build()
+    val struct2 = StructureShape
+      .builder()
+      .id(struct2ShapeId)
+      .addMember(structMember2)
+      .build()
+
+    val model =
+      Model.builder().addShapes(struct, union, struct2).build()
+
+    val result = validator.validate(model).asScala.toList
+
+    val expected = List(
+      ValidationEvent
+        .builder()
+        .id("AdtMemberTrait")
+        .shape(struct)
+        .severity(Severity.ERROR)
+        .message(
+          "test#struct is improperly referenced from [test#MyStruct2]"
+        )
+        .build()
+    )
+    expect(result == expected)
+  }
+
+  test("return error when structure has no members") {
```

Comment:
Can you elaborate on why this check is needed ?

Reply:
At first I though that the rule was to ensure that the _union_ had atleast a member, but its not. I generated a sample and it looks like that:

```scala
package smithy4s.example

import smithy4s.schema.Schema._

sealed trait OrderType2 extends scala.Product with scala.Serializable
object OrderType2 extends smithy4s.ShapeTag.Companion[OrderType2] {
  val id: smithy4s.ShapeId = smithy4s.ShapeId("smithy4s.example", "OrderType2")

  val hints : smithy4s.Hints = smithy4s.Hints.empty

  case class NameCase(name: String) extends OrderType2
  case class OtherStoreOrder() extends OrderType2
  object OtherStoreOrder extends smithy4s.ShapeTag.Companion[OtherStoreOrder] {
    val id: smithy4s.ShapeId = smithy4s.ShapeId("smithy4s.example", "OtherStoreOrder")

    val hints : smithy4s.Hints = smithy4s.Hints.empty

    implicit val schema: smithy4s.Schema[OtherStoreOrder] = constant(OtherStoreOrder()).withId(id).addHints(hints)

    val alt = schema.oneOf[OrderType2]("inStore")
  }

  object NameCase {
    val hints : smithy4s.Hints = smithy4s.Hints.empty
    val schema: smithy4s.Schema[NameCase] = bijection(string.addHints(hints), NameCase(_), _.name)
    val alt = schema.oneOf[OrderType2]("name")
  }

  implicit val schema: smithy4s.Schema[OrderType2] = union(
    NameCase.alt,
    OtherStoreOrder.alt,
  ){
    case c : NameCase => NameCase.alt(c)
    case c : OtherStoreOrder => OtherStoreOrder.alt(c)
  }.withId(id).addHints(hints)
}
```

Reply:
Yeah, upon further inspection this isn't necessary. I'll remove it